src/controllers/user.js

```diff
@@ -57,11 +57,59 @@ export const getAll = async (req, res) => {
 }
 
 export const updateById = async (req, res) => {
-  const { cohort_id: cohortId } = req.body
+  const id = Number(req.params.id)
+  const { firstName, lastName, bio, githubUrl, cohortId, profilePicture } =
+    req.body
 
-  if (!cohortId) {
-    return sendDataResponse(res, 400, { cohort_id: 'Cohort ID is required' })
-  }
+  try {
+    // Check user you want to update exists:
+    const foundUser = await User.findById(id)
+
+    if (!foundUser) {
+      return sendDataResponse(res, 404, { error: 'User not found' })
+    }
+
+    // Check whether user is authorised
+    const canUpdateProfile = req.user.id === id || req.user.role === 'TEACHER'
 
-  return sendDataResponse(res, 201, { user: { cohort_id: cohortId } })
+    if (canUpdateProfile === false) {
+      return res
+        .status(403)
+        .json({ error: 'User not authorized to make this change.' })
+    }
+
+    if (req.user.id === id) {
+      if (!firstName || !lastName) {
+        return sendDataResponse(res, 400, {
+          error: 'First name and Last name is required'
+        })
+      }
+      // inject fields if not null in payload
+      const updateData = {
+        firstName,
+        lastName,
+        ...(bio && { bio }),
+        ...(githubUrl && { githubUrl }),
+        ...(profilePicture && { profilePicture })
+      }
+      const updatedUser = await User.updateUser(id, updateData)
+      delete updatedUser.password
+      return sendDataResponse(res, 201, updatedUser)
+    }
+
+    if (req.user.role === 'TEACHER') {
```

Comment:
you're saying if I am a teacher, then I can only update the cohortId of a user? what if a user writes offensive text and a teacher wants to update / purge the profile?

I think STUDENT + TEACHER should be able to fully update the profile; except TEACHER can change cohortId on top

Reply:
Let me know if you decide with this logic anyways :)